internal/handler/handler.go

```diff
@@ -65,31 +88,243 @@ func New(opts *NewOpts) (*Handler, error) {
 		b = oplog.NewBackend(b, opts.L.Named("oplog"))
 	}
 
+	if opts.CappedCleanupPercentage >= 100 || opts.CappedCleanupPercentage <= 0 {
+		return nil, fmt.Errorf(
+			"percentage of documents to cleanup must be in range (0, 100), but %d given",
+			opts.CappedCleanupPercentage,
+		)
+	}
+
 	h := &Handler{
 		b:       b,
 		NewOpts: opts,
 		cursors: cursor.NewRegistry(opts.L.Named("cursors")),
+
+		cappedCleanupStop: make(chan struct{}),
+		cleanupCappedCollectionsDocs: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: namespace,
+				Subsystem: subsystem,
+				Name:      "cleanup_capped_docs",
+				Help:      "Total number of documents deleted in capped collections during cleanup.",
+			},
+			[]string{"db", "collection"},
+		),
+		cleanupCappedCollectionsBytes: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: namespace,
+				Subsystem: subsystem,
+				Name:      "cleanup_capped_bytes",
+				Help:      "Total number of bytes freed in capped collections during cleanup.",
+			},
+			[]string{"db", "collection"},
+		),
 	}
 
 	h.initCommands()
 
+	h.wg.Add(1)
+
+	go func() {
+		defer h.wg.Done()
+
+		h.runCappedCleanup()
+	}()
```

Comment:
```suggestion
```

Reply:
I'd consider exposing a `StartCappedCleanup` or `RunCappedCleanup` function which you'd use as following as it might come as a surprise that a New function is starting a goroutine. It also might make it easier to test some internal details, if we ever want to.

```go
go r.RunCappedCleanup(ctx)
```

Reply:
Yeah, we discussed just that, but decided to keep it as is for now